src/CheckersBoard.java

```diff
@@ -1,553 +1,403 @@
-		}
-	}//end move
-	
-	public int getOCount(){ return oCount; }
-	
-	public int getXCount(){ return xCount; }
-	
-	
-	/**A toString method for the CheckersBoard, allows easier access for console based interface
-	 * @return result The result of the toString(), aka the gameBoard
-	 */
-	public String toString(){//prints out the checkerBoard
-		String result;
-		result = " | A | B | C | D | E | F | G | H |";
-		
-		for(int i = 0; i < 32; i++){
-			if(i%4 == 0){
-				result += "\n-+---+---+---+---+---+---+---+---+\n";
-			}
-			
-			if(i == 0){
-				result += "1";
-			}else if(i == 4){
-				result += "2|";
-			}else if(i == 8){
-				result += "3";
-			}else if(i == 12){
-				result += "4|";
-			}else if(i == 16){
-				result += "5";
-			}else if(i == 20){
-				result += "6|";
-			}else if(i == 24){
-				result += "7";
-			}else if(i == 28){
-				result += "8|";
-			}
+    public char getWinner() { return winner; }
+
+    public Boolean getError() { return error; }
+    
+    public void changeTurn() {
+	if (turn == 'x') {
+	    turn = 'o';
+	}
+	else {
+	    turn = 'x';
+	}
+    }
 
-			if(i < 4){
-				result += "|   | " + pieces.get(i) + " ";
-			}else if(i < 8){
-				result += " " + pieces.get(i) + " |   |";
-			}else if(i < 12){
-				result += "|   | " + pieces.get(i) + " ";
-			}else if(i < 16){
-				result += " " + pieces.get(i) + " |   |";
-			}else if(i < 20){
-				result += "|   | " + pieces.get(i) + " ";
-			}else if(i < 24){
-				result += " " + pieces.get(i) + " |   |";
-			}else if(i < 28){
-				result += "|   | " + pieces.get(i) + " ";
-			}else if(i < 32){
-				result += " " + pieces.get(i) + " |   |";
-			}
+    public int getXCount() { return xCount; }
 
-			if(i == 3 || i == 11 || i == 19 || i == 27){
-				result += "|";
-			}
-		}//end for
-		result += "\n-+---+---+---+---+---+---+---+---+\n";
-		return result;
+    public int getOCount() { return oCount; }
+    
+    /**Checks if the index you are moving from is owned by the correct owner
+     * @param i,j The index of the spot you are trying to move
+     * @return True if you own the spot, or False if you do not own it
+     */
+    public boolean correctOwner(int i, int j) {
+	if (turn == 'x') {
+	    if(pieces[i][j] == 'x' || pieces[i][j] == 'X') {
+		return true;
+	    }
+	    else {
+		return false;
+	    }
+	}
+	else if (turn == 'o') {
+	    if (pieces[i][j] == 'o' || pieces[i][j] == 'O') {
+		return true;
+	    }
+	    else {
+		return false;
+	    }
+	}
+	else {
+	    return false;
+	}
+    }
 
+    /** move method that allows the user to move
+     * Will throw a CheckersIllegalMoveException if the from index is a spot that you don't own, or if you are trying to move to a spot that is not valid
+     *  @param from the integer value of the index of the spot you are moving from
+     *  @param to the integer value of the index of the spot you are moving to
+     */
+    public void setMoves(int fromI, int fromJ, int toI, int toJ) {
+        if (turn == 'x') {
```

Comment:
What if you just set a "y direction" variable in the condition (to -1 or +1) and then multiply the things that matter?